### PR TITLE
8289780: Avoid formatting stub names when Forte is not enabled

### PR DESCRIPTION
**Pull request by iklam**
`Forte::register_stub()` should be called only when the JVm is being instrumented by Forte (aka "Oracle Developer Studio")

https://www.oracle.com/tools/developerstudio/downloads/developer-studio-jsp.html

We currently always format the name of generated stubs and call `Forte::register_stub()`, which usually does nothing.

Example:

https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/sharedRuntime.cpp#L2686-L2697

To improve start-up, we should check if Forte is enabled before formatting the name.

I also renamed some `#ifndef IA64` around the code that I touched.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289780](https://bugs.openjdk.org/browse/JDK-8289780): Avoid formatting stub names when Forte is not enabled

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [a1523c9d](https://git.openjdk.org/jdk/pull/9386/files/a1523c9dcbba5c270ada55015986be67a0d5c6c0)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9386/head:pull/9386` \
`$ git checkout pull/9386`

Update a local copy of the PR: \
`$ git checkout pull/9386` \
`$ git pull https://git.openjdk.org/jdk pull/9386/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9386`

View PR using the GUI difftool: \
`$ git pr show -t 9386`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9386.diff">https://git.openjdk.org/jdk/pull/9386.diff</a>

</details>

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Cleanup looks good! Thanks.

**Comment by iklam** [comment]
It turns out that `collector_func_load` is used by OracleDeveloperStudio on Linux as well, so I can't remove the call to collector_func_load. I think this (weak) function will be non-null when OracleDeveloperStudio loads the JVM:

```
$ nm OracleDeveloperStudio12.6-linux-x86-bin/developerstudio12.6/lib/analyzer/amd64/libcollector.so \
     | grep collector_func_load
000000000002efd0 T __collector_func_load
000000000002efd0 W collector_func_load
```

So slight change of plan. I am adding a check to avoid computing the `blob_id`, etc, when Forte::is_enabled()` is false, which will be the case in most cases.

File: src/hotspot/share/prims/forte.hpp [hunk 1]
```diff
@@ -29,6 +29,7 @@
 
 class Forte : AllStatic {
  public:
+   static bool is_enabled() NOT_JVMTI_RETURN_(false);
```
Comment (coleenp) [inline src/hotspot/share/prims/forte.hpp:32]: I don't think the rest of this forte code is disabled by JVMTI.

Comment (coleenp) [reply src/hotspot/share/prims/forte.hpp:32]: If the answer to whether it's enabled is something you want to be fast, and doesn't change, maybe make it check a variable?

Comment (iklam) [reply src/hotspot/share/prims/forte.hpp:32]: The code in the `Forte` class, as well as the non-trivial of `AsyncGetCallTrace()` in forte.cpp, are inside `#if INCLUDE_JVMTI`. That's why I use `NOT_JVMTI_RETURN_` for the new function.

Comment (coleenp) [reply src/hotspot/share/prims/forte.hpp:32]: Ok, thanks for answering my question.

**Review by coleenp — APPROVED** [review]

File: src/hotspot/share/runtime/sharedRuntime.cpp [hunk 2]
```diff
@@ -2684,16 +2684,20 @@ extern "C" void unexpected_adapter_call() {
...
+                 "%s(%s)",
+                 new_adapter->name(),
+                 entry->fingerprint()->as_string());
+    if (Forte::is_enabled()) {
+      Forte::register_stub(blob_id, new_adapter->content_begin(), new_adapter->content_end());
+    }
+
+    if (JvmtiExport::should_post_dynamic_code_generated()) {
+      JvmtiExport::post_dynamic_code_generated(blob_id, new_adapter->content_begin(), new_adapter->content_end());
+    }
```
Comment (sspitsyn) [inline src/hotspot/share/runtime/sharedRuntime.cpp:2700]: The lines 2698-2670 is better to move out of the if-statement at the line 2687.

**Review by sspitsyn — APPROVED** [review]
I've posed one comment.
Other than that the fix looks okay.
Thanks,
Serguei

File: src/hotspot/share/runtime/sharedRuntime.cpp (hunk 2 above)
Comment (iklam) [reply src/hotspot/share/runtime/sharedRuntime.cpp:2700]: HI Serguei, thanks for the review. Lines 2698-2700 need the blob_id which is generated at line 2688, so they have to stay inside the outer "if" block.

Comment (sspitsyn) [reply src/hotspot/share/runtime/sharedRuntime.cpp:2700]: Thanks, Ioi. You are right.

**Comment by iklam** [comment]
Thanks @dholmes-ora, @sspitsyn, @coleenp for the review.
/integrate

**Pull request closed without merging**